lib/typhoeus/response/status.rb

```diff
@@ -81,6 +92,16 @@ def first_header_line
           end
         end
       end
+
+      # :nodoc:
+      def has_good_response_code?
+        response_code >= 200 && response_code < 300
```

Comment:
This doesn't feel "right"

Anything 4xx (your fault) and 5xx (my fault) is erroneous. 3xx doesn't really feel like something went "wrong", but instead it's more along the lines of "you need to handle this case" (redirects, not modified, etc.)

@i0rek what do you think?